contracts/src/common/hca/HCAEquivalence.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import {IHCAFactoryBasic} from "./interfaces/IHCAFactoryBasic.sol";
+
+/// @dev Replaces msg.sender
+abstract contract HCAEquivalence {
```

Comment:
Couldn't this be part of `HCAContext` directly? When do you need this but not `Context`?